x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/input/CachedBlobContainerIndexInput.java

```diff
@@ -36,7 +35,7 @@ public class CachedBlobContainerIndexInput extends MetadataCachingIndexInput {
      * a complete part of the {@link #fileInfo} at once in the cache and should not be
      * used for anything else than what the {@link #prefetchPart(int, Supplier)} method does.
      */
-    public static final IOContext CACHE_WARMING_CONTEXT = new IOContext(IOContext.Context.DEFAULT, null, null, ReadAdvice.NORMAL);
+    public static final IOContext CACHE_WARMING_CONTEXT = IOContext.DEFAULT.withHints();
```

Comment:
is `withHints` needed here? it'll return the same context, no?

Reply:
This is just so a new object is created for reference equality checks elsewhere. Removing this is done by #127575

Reply:
Actually, this does still need to be here...